package.json

```diff
@@ -1,232 +1,57 @@
-    "node-sass": "4.9.3",
-    "npmlog": "4.1.2",
-    "opn": "5.3.0",
-    "ora": "2.1.0",
-    "os-locale": "2.1.0",
-    "parse-json": "4.0.0",
-    "pkg-up": "2.0.0",
-    "pretty-ms": "4.0.0",
-    "promptly": "2.2.0",
-    "propose": "0.0.5",
-    "pump": "3.0.0",
-    "remix-debug-debugtest": "0.2.16",
-    "remix-tests": "0.1.1",
-    "request": "2.88.0",
-    "sass-loader": "7.1.0",
-    "semver": "5.5.1",
-    "serve-static": "1.13.2",
-    "shelljs": "0.5.3",
-    "simples": "0.8.8",
-    "solc": "0.5.0",
-    "solidity-parser-antlr": "0.3.2",
-    "source-map-support": "0.5.9",
-    "stream-json": "1.1.3",
-    "string-replace-async": "1.2.1",
-    "style-loader": "0.23.1",
-    "swarm-api": "0.1.2",
-    "tar": "3.2.1",
-    "term.js": "0.0.7",
-    "url-loader": "1.1.2",
-    "uuid": "3.3.2",
-    "viz.js": "1.8.2",
-    "vm2": "3.6.4",
-    "web3": "1.0.0-beta.37",
-    "web3-bzz": "1.0.0-beta.37",
-    "web3-core": "1.0.0-beta.37",
-    "web3-core-helpers": "1.0.0-beta.37",
-    "web3-core-method": "1.0.0-beta.37",
-    "web3-core-promievent": "1.0.0-beta.37",
-    "web3-core-requestmanager": "1.0.0-beta.37",
-    "web3-core-subscriptions": "1.0.0-beta.37",
-    "web3-eth": "1.0.0-beta.37",
-    "web3-eth-abi": "1.0.0-beta.37",
-    "web3-eth-accounts": "1.0.0-beta.37",
-    "web3-eth-contract": "1.0.0-beta.37",
-    "web3-eth-iban": "1.0.0-beta.37",
-    "web3-eth-personal": "1.0.0-beta.37",
-    "web3-net": "1.0.0-beta.37",
-    "web3-providers-http": "1.0.0-beta.37",
-    "web3-providers-ipc": "1.0.0-beta.37",
-    "web3-providers-ws": "1.0.0-beta.37",
-    "web3-shh": "1.0.0-beta.37",
-    "web3-utils": "1.0.0-beta.37",
-    "webpack": "4.19.0",
-    "window-size": "1.1.1",
-    "ws": "6.1.2"
-  },
   "devDependencies": {
-    "@babel/cli": "7.1.2",
-    "@babel/plugin-proposal-optional-chaining": "7.0.0",
-    "@types/async": "2.0.50",
-    "@types/body-parser": "1.17.0",
-    "@types/cors": "2.8.4",
-    "@types/express": "4.16.0",
-    "@types/express-http-proxy": "1.5.0",
-    "@types/express-ws": "3.0.0",
-    "@types/globule": "1.1.3",
-    "@types/handlebars": "4.0.39",
-    "@types/helmet": "0.0.42",
-    "@types/i18n": "0.8.3",
-    "@types/node": "10.11.7",
-    "@types/os-locale": "2.1.0",
-    "@types/pretty-ms": "3.2.0",
-    "@types/request": "2.48.1",
-    "@types/web3": "1.0.12",
-    "babel-plugin-dynamic-import-node": "2.2.0",
-    "chai": "4.1.2",
-    "chalk": "2.4.1",
-    "cross-env": "5.2.0",
-    "eslint": "5.7.0",
-    "eslint-plugin-react": "7.11.1",
+    "chalk": "2.4.2",
+    "lerna": "3.10.7",
     "minimist": "1.2.0",
-    "mocha-sinon": "1.2.0",
     "npm-run-all": "4.1.5",
-    "nyc": "13.1.0",
-    "rimraf": "2.6.2",
-    "sinon": "4.5.0",
-    "standard-version": "4.4.0",
-    "tslint": "5.11.0",
-    "typescript": "3.1.3"
+    "rimraf": "2.6.3",
+    "semver": "5.6.0"
   },
   "engines": {
-    "node": ">=8.11.3",
-    "npm": ">666",
+    "node": ">=8.12.0",
+    "npm": ">=6.4.1",
```

Comment:
It's important to use yarn at the top-level since [`"workspaces"`](https://github.com/embark-framework/embark/blob/build/lerna/package.json) is not supported by npm.

npm is still used in various ways in the development setup, and since v5 is pretty buggy, I pegged it to the first version 6.x npm that got bundled with a node release, i.e. node v8.12.0. Given when/how npm enforces the engines check with respect to itself, it's more of a "signal to humans" than an automatic protection. In any case, it only applies to developers working on embark itself.

Note that owing to the monorepo setup, it becomes impossible to install embark directly from github since `hosted-git-info` (used internally by npm) doesn't have a mechanism for identifying a specific package within the directory structure of a git repository. Yarn doesn't offer support for that either. It's a trade-off — nice capability, sad to lose it, but the benefits of the monorepo will hopefully outweigh the loss.

Reply:
Maybe we can adapt #1263 so that if a developer tries to `npm install` there will be an error.

Reply:
I looked into the technique of #1263 but it won't work. There doesn't seem to be a way, via environment variables available to the `preinstall` script, to distinguish `npm install` in the repo vs. `npm install [pkg]`, where `[pkg]` is a tarball path or identifier on the registry or was listed as a dev/Dep in a `package.json`.

Since we want users to be able to install embark from the registry using npm, I think it's a no-go. We'll just need to educate developers who want to work on embark or run a development build of embark that they need to `yarn install` in the top-level of the cloned repo instead of doing `npm install`.

Reply:
Yes, personally I think that is probably the right approach. Since `npm install`ing embark from the registry is OK, and only developers wishing to contribute to embark are required to use yarn, adding some information where necessary (on the README and website) would be sufficient.